src/realm/string_data.hpp

```diff
@@ -278,6 +278,7 @@ inline bool operator<(const StringData& a, const StringData& b) noexcept
         // equal to empty strings.
         return true;
     }
+    // FIXME: This should be using `std::char_traits<char>::compare(...)` or something similar.
```

Comment:
Do you want to introduce a fixme or did you want to fix it in this PR?

Reply:
@bmunkholm , fixing this in this PR would require a large file-format change since this would change the sort order of all string columns. Maybe this can just be a regular comment to warn that this function is subtly broken. I think `std::char_traits<char>::compare()` would actually also be broken when comparing against std::string/std::string_view's because std::string's operator< takes the size of the strings into account separately from the result of `std::char_traits<char>::compare()`

Reply:
Could we add this explanation in the FIXME, so others that come across this will understand how it's broken and why we haven't fixed it yet? Should we even make an issue (with label Breaking) to capture this, so that when we are doing a breaking update next time we could do this as well?